wallet-new/integration/Functions.hs

```diff
@@ -35,454 +37,455 @@ import qualified Pos.Wallet.Web.ClientTypes.Types as V0
-  where
-    generateNewWallet spendPass =
-        NewWallet
-            <$> arbitrary
-            <*> pure (Just spendPass)
-            <*> arbitrary
-            <*> pure "Wallet"
-            <*> pure CreateWallet
+            -- Modify wallet state accordingly.
+            wallets    <>= [result]
+            walletsPass . at result ?= newPassword
+          where
+            generateNewWallet spendPass =
+                NewWallet
+                    <$> arbitrary
+                    <*> pure (Just spendPass)
+                    <*> arbitrary
+                    <*> pure "Wallet"
+                    <*> pure CreateWallet
 
-runAction wc ws GetWallets   = do
-    -- We choose from the existing wallets.
-    result  <-  respToRes $ getWallets wc
+        GetWallets -> do
+            localWallets <- use wallets
+            -- We choose from the existing wallets.
+            result  <-  respToRes $ getWallets wc
 
-    checkInvariant
-        (length result == length (ws ^. wallets))
-        (LocalWalletsDiffers result)
+            checkInvariant
+                (length result == length localWallets)
+                (LocalWalletsDiffers result localWallets)
 
-    -- No modification required.
-    pure $ ws
-        & actionsNum +~ 1
+            -- No modification required.
 
-runAction wc ws GetWallet    = do
-    -- We choose from the existing wallets.
-    wallet  <-  pickRandomElement (ws ^. wallets)
-    result  <-  respToRes $ getWallet wc (walId wallet)
+        GetWallet -> do
+            ws <- get
 
-    checkInvariant
-        (walBalance result == minBound)
-        (LocalWalletDiffers result)
+            let localWallets = ws ^. wallets
 
-    -- No modification required.
-    pure $ ws
-        & actionsNum +~ 1
+            -- The precondition is that we need to have wallets.
+            guard (not (null localWallets))
 
-runAction wc ws DeleteWallet = do
+            -- We choose from the existing wallets.
+            wal@Wallet{ walId }  <-  pickRandomElement localWallets
 
-    let localWallets = ws ^. wallets
+            log $ "Requesting wallet: " <> show walId
+            result  <-  respToRes $ getWallet wc walId
 
-    -- The precondition is that we need to have wallets.
-    guard (not (null localWallets))
+            checkInvariant
+                (walBalance result == minBound)
+                (LocalWalletDiffers result wal)
 
-    -- We choose from the existing wallets.
-    wallet  <-  pickRandomElement localWallets
+            -- No modification required.
 
-    -- If we don't have any http client errors, the delete was a success.
-    _       <-  either throwM pure =<< deleteWallet wc (walId wallet)
+        DeleteWallet -> do
+            localWallets <- use wallets
 
-    -- Just in case, let's check if it's still there.
-    result  <-  respToRes $ getWallets wc
+            -- The precondition is that we need to have wallets.
+            guard (not (null localWallets))
```

Comment:
we don't even need to have this guard now, because `pickRandomElement` bellow in its current form will be the guard.